documentation/src/main/asciidoc/topics/query.adoc

```diff
@@ -18,480 +18,570 @@ Apart from indexed queries, {brandname} can run queries over non-indexed or
 partially indexed data.
 
 In terms of Search APIs, {brandname} has its own query language called _link:#query_ickle[Ickle]_, which is string-based and adds support for full-text
-querying. The link:#query_dsl[Query DSL] can be used for both embedded and remote java clients when full-text is not required; for Java embedded clients
-{brandname} offers the link:#query_hibernatesearch[Hibernate Search Query API] which supports running Lucene queries in the grid, apart from advanced search capabilities
-like Faceted and Spatial search.
+querying.
 
 Finally, {brandname} has support for link:#query_continuous[Continuous Queries], which works in a reverse manner to the other APIs: instead of creating, executing a query
 and obtain results, it allows a client to register queries that will be evaluated continuously as data in the cluster changes, generating notifications
 whenever the changed data matches the queries.
 
-[[query_library]]
-== Embedded Querying
+== Indexing
 
-Embedded querying is available when {brandname} is used as a library. No protobuf mapping is required, and both indexing and searching
-are done on top of Java objects. When in library mode, it is possible to run Lucene queries directly and use all the available link:#query_apis[Query APIs] and it also allows flexible indexing configurations to keep latency to a minimal.
+Indexing in {brandname} happens on a per-cache basis and by default a cache is not indexed. Enabling indexing is not mandatory but queries using an index will
+have a vastly superior performance. On the other hand, enabling indexing can impact negatively the write throughput of a cluster, so make sure to check the link:#query_performance[query performance guide] for some strategies to minimize this impact depending on the cache type and use case.
 
-=== Quick example
+NOTE: When using the full-text operator (:) or any other full-text feature, indexing is mandatory. {brandname} will
+throw an error if a full-text query is attempted in a non-indexed cache.
 
-We're going to store _Book_ instances in an {brandname} cache called "books". _Book_ instances will be indexed, so we enable indexing for the cache,
-letting {brandname} link:#query_autoconfig[configure the indexing automatically]:
+=== Configuration
 
-{brandname} configuration:
+To enable indexing via XML, you need to add the `<indexing>` element to your cache configuration, specify the entities that are indexed and optionally pass additional properties.
+
+NOTE: The presence of an `<indexing>` element which omits the `enabled` attribute will auto-enable indexing for your
+convenience, even though the default value of the `enabled` attribute is defined as `"false"` in the XSD schema.
+This behaviour exists only on schemas starting with version 11.
```

Comment:
One general comment is that admonitions should take the format of:

[NOTE]
====
Note text goes here.
====

Also this note seems to add complexity to a shortcut for users. Why don't we update the attribute to "true" in the XSD?

As this doc is going to be merged for v11 I don't think we need that last sentence.

Reply:
Let's not change the config now :)